Drug data

aditnath opened the pull request:
Initial commit of the drug dataset for viewing on the cell feature explorer

meganrm commented on data/dataset-drug-0-1/dataset.json:
```diff
@@ -0,0 +1,36 @@
+    "totalTaggedStructures": 1
+  },
+  "featureDefsPath": "feature_defs.json",
+  "featuresDataPath": "cell_feature_analysis.json",
```
this needs to match your file name, right now it's `feature_analysis.json`. So you either need to chance this value or change the filename

meganrm commented:
I'm not sure why it says pass but you still have an error: https://github.com/allen-cell-animated/cell-feature-data/actions/runs/4703356799/jobs/8341747389. 

The cell_feature_analysis file does not have correctly formatted data in it either. for each cell it should have an object that is has two attributes: {`measured_features`; [], `file_info`: []}. 

Measured_features is the list of values you want to be able to plot, ie it's the values for each of the items in `feature_defs`. `file_info` is for per cell data that has to do with looking up where the image data is stored:            "description": "Ordered array of file info: [id, parentId, id of default group, thumbnail path, volume viewer path, parent thumbnail path, parent volume viewer path]",

Any of these can be empty strings if they don't apply.

toloudis commented:
`file_info` is for per cell data that has to do with looking up where the image data is stored: "description": "Ordered array of file info: [id, parentId, id of default group, thumbnail path, volume viewer path, parent thumbnail path, parent volume viewer path]",
> 
> Any of these can be empty strings if they don't apply.

I can tell you what form the volume viewer path and thumbnail path need to take.  parent volume viewer path will be empty.  I just need togo back and look at how I named the generated files.

toloudis commented:
thumbnail path is of the form "original filename.czi.png" like this:  `3500001171_100X_20170811_1-Scene-1-P7-D05.czi.png`
volume-viewer path is of the form: "original filename.czi.zarr" like this `3500001171_100X_20170811_1-Scene-1-P7-D05.czi.zarr`
The base url directories for these go into the dataset.json file...
For thumbnails it is 
`https://s3.console.aws.amazon.com/s3/buckets/animatedcell-test-data/drug_pilot/thumbs`
and for images it is 
`https://s3.console.aws.amazon.com/s3/buckets/animatedcell-test-data/drug_pilot/images`

aditnath commented:
@toloudis or @meganrm do you have a script that takes a csv with all relevant fields and converts it to the correct format needed for cell_feature_analysis.json?

toloudis commented:
> @aditnath thumbnail path is of the form "original filename.czi.png" like this: `3500001171_100X_20170811_1-Scene-1-P7-D05.czi.png` volume-viewer path is of the form: "original filename.czi.zarr" like this `3500001171_100X_20170811_1-Scene-1-P7-D05.czi.zarr` The base url directories for these go into the dataset.json file... For thumbnails it is `https://s3.console.aws.amazon.com/s3/buckets/animatedcell-test-data/drug_pilot/thumbs` and for images it is `https://s3.console.aws.amazon.com/s3/buckets/animatedcell-test-data/drug_pilot/images`

@aditnath You still need to change thumbnailRoot and volumeViewerDataRoot in dataset.json as mentioned above.

I also think it might be a problem that the fileInfo has empty ID values for all the images.  I'm fairly sure ID is required but not sure what we should use for IDs.  (@meganrm do you think they need to be short or could they be the filename without extensions?)

meganrm commented:
> > @aditnath thumbnail path is of the form "original filename.czi.png" like this: `3500001171_100X_20170811_1-Scene-1-P7-D05.czi.png` volume-viewer path is of the form: "original filename.czi.zarr" like this `3500001171_100X_20170811_1-Scene-1-P7-D05.czi.zarr` The base url directories for these go into the dataset.json file... For thumbnails it is `https://s3.console.aws.amazon.com/s3/buckets/animatedcell-test-data/drug_pilot/thumbs` and for images it is `https://s3.console.aws.amazon.com/s3/buckets/animatedcell-test-data/drug_pilot/images`
> 
> @aditnath You still need to change thumbnailRoot and volumeViewerDataRoot in dataset.json as mentioned above.
> 
> I also think it might be a problem that the fileInfo has empty ID values for all the images. I'm fairly sure ID is required but not sure what we should use for IDs. (@meganrm do you think they need to be short or could they be the filename without extensions?)

They are definitely required. They are used both to store the per cell data, to label the cells in the gallery, and to request the volumetric data. So yeah, in this case they should be the file name without the extension

aditnath commented:
> @aditnath thumbnail path is of the form "original filename.czi.png" like this: `3500001171_100X_20170811_1-Scene-1-P7-D05.czi.png` volume-viewer path is of the form: "original filename.czi.zarr" like this `3500001171_100X_20170811_1-Scene-1-P7-D05.czi.zarr` The base url directories for these go into the dataset.json file... For thumbnails it is `https://s3.console.aws.amazon.com/s3/buckets/animatedcell-test-data/drug_pilot/thumbs` and for images it is `https://s3.console.aws.amazon.com/s3/buckets/animatedcell-test-data/drug_pilot/images`

The thumbnail root s3 path goes the the field "thumbnailRoot" I imagine, but what does the images path go to. Both "downloadRoot" and "volumeViewerDataRoot" exist, and not sure which one takes the image path

toloudis commented:
> 
> What goes in "downloadRoot"? Should i leave it unchanged or blank?

For this dataset download will not be an option so that can be blank.

toloudis commented:
> @toloudis and @meganrm I added ids to all files in cell_feature_analysis.json, and corrected all the dataset.json s3 paths. Let me know if things look ok

One thing I noticed with a quick check is that several rows seem to be identical.  Something seems to be wrong there - I am pretty sure we are intending to release one row per FOV, but for whatever reason I expected the names in each row to be unique.

aditnath commented:
> > @toloudis and @meganrm I added ids to all files in cell_feature_analysis.json, and corrected all the dataset.json s3 paths. Let me know if things look ok
> 
> One thing I noticed with a quick check is that several rows seem to be identical. Something seems to be wrong there - I am pretty sure we are intending to release one row per FOV, but for whatever reason I expected the names in each row to be unique.

https://allencellscience.slack.com/archives/C04QUD7UYRH/p1677096434964089
I was using the csv from the slack message linked above to create cell_feature_analysis.json, is that not correct?

toloudis commented:
> > One thing I noticed with a quick check is that several rows seem to be identical. Something seems to be wrong there - I am pretty sure we are intending to release one row per FOV, but for whatever reason I expected the names in each row to be unique.
> 
> https://allencellscience.slack.com/archives/C04QUD7UYRH/p1677096434964089 I was using the csv from the slack message linked above to create cell_feature_analysis.json, is that not correct?

I thought we were releasing only FOV images... it is just a couple messages further down that thread:
https://allencellscience.slack.com/archives/C04QUD7UYRH/p1677174482202359

We can revisit that decision with Nathalie, Antoine and others, but if we are going to release single cell images then I will redo the image processing and we obviously need unique IDs, not FOV filename IDs.  ID is something that should be unique per row.

aditnath commented:
> > > One thing I noticed with a quick check is that several rows seem to be identical. Something seems to be wrong there - I am pretty sure we are intending to release one row per FOV, but for whatever reason I expected the names in each row to be unique.
> > 
> > 
> > https://allencellscience.slack.com/archives/C04QUD7UYRH/p1677096434964089 I was using the csv from the slack message linked above to create cell_feature_analysis.json, is that not correct?
> 
> I thought we were releasing only FOV images... it is just a couple messages further down that thread: https://allencellscience.slack.com/archives/C04QUD7UYRH/p1677174482202359
> 
> We can revisit that decision with Nathalie, Antoine and others, but if we are going to release single cell images then I will redo the image processing and we obviously need unique IDs, not FOV filename IDs. ID is something that should be unique per row.

Makes total sense to just release FOV images, would I need to only add unique rows in the csv then?

toloudis commented:
> Makes total sense to just release FOV images, would I need to only add unique rows in the csv then?
Yes I believe so!

toloudis commented:
@aditnath
"totalCells" was missing from the userData and I added it.  It probably should be optional but turned out to be required by CFE.

But then featuresDataOrder in dataset.json has 9 items but there are only 7 features in the actual feature data (Our validation scripts should be catching this for sure).  If you just change that featuresDataOrder to be consistent then that will fix the next error.

toloudis commented:
> @toloudis updated the PR with the fixes (actually was missing a feature, added it so now there are 8 features, and that matches in all the json files). Should I push the dataset to staging again?

Yes - at this point you can keep iterating in this way.  If the validation action is green, then you can run the update-to-staging action.  If that succeeds, then you can try loading it in staging.cfe.allencell.org.  If the last step looks broken, I can help diagnose
